week2/main.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+)
+
+func RemoveSpecialCharacter(text string) string {
+
+	// Make a Regex to say we only want letters and numbers
+	reg, err := regexp.Compile("[^a-zA-Z0-9]+")
+	if err != nil {
+		log.Fatal(err)
+	}
+	validCharacter := reg.ReplaceAllString(text, " ")
+
+	return validCharacter
+}
+
+//Read the content of files and remove special character
+func ReadContentByFile(file_name string, channel chan []string) {
+	file, err := os.Open(file_name)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer file.Close()
+
+	reader := bufio.NewReader(file)
+	content, _ := ioutil.ReadAll(reader)
+	validate_text := RemoveSpecialCharacter(string(content))
+	valid_string := strings.Split(validate_text, " ")
+
+	fmt.Printf("Valid text: %s\n", valid_string)
+	channel <- valid_string
+}
+
+//Count ocurrency of data
+func CountOfOccurences(data []string, counts map[string]int) {
+	for _, i := range data {
+		counts[i]++
+	}
+}
+
+func main() {
+	//Create list to store for all file name
+	var InputFileList []string
+
+	//Path of files
+	InputFilePath := "./DataTest"
+
+	//creating chanel to count
+	counts := make(map[string]int)
+	//channel between routines
+	channel := make(chan []string)
+
+	//Create files from the directory
+	files, err := ioutil.ReadDir(InputFilePath)
```

Comment:
You should move this part into a new function, it'll make you code cleaner. You may like to read this article https://code.tutsplus.com/tutorials/solid-part-1-the-single-responsibility-principle--net-36074

Reply:
Thanks for your advice. Let me try to improve it by this article.